eureka-core/src/main/java/com/netflix/eureka2/interests/NotificationsSubject.java

```diff
@@ -101,6 +100,13 @@ public ResumeResult resume() {
         }
     }
 
+    protected void drainBuffer() {
+        ChangeNotification<T> nextPolled;
+        while ((nextPolled = notificationsWhenPaused.poll()) != null) { // drain pending queue.
+            notificationSubject.onNext(nextPolled); // Since pause flag is not yet unset, don't call this.onNext()
```

Comment:
This comment is outdated now that this is in a separate method. Are we only ever allowed to drainBuffer() as part of resume or if we are not currently paused? Can we add a comment here to details these restrictions?